docs/adr/adr_001_cache_bust.md

```diff
@@ -0,0 +1,322 @@
+
+![](adr_001_sign_retry_errors.png)
+
+**Efforts**
+
+Medium. We may have users reporting indirect misbehaviours because of this.
+
+**Rollout**: N/A
+
+### Option 1 - Bump data timestamp on signature refresh
+
+This option consists in changing the server code when a request to refresh the collection signature is received.
+Currently, when a signature is refreshed, since the data itself isn't modified, we only refresh the collection metadata timestamp.
+
+With this option, we will bump the data timestamp, to pretend there was data change.
+
+The timestamp in the monitor/changes response will thus be updated. The highest data timestamp in the list will then also be bumped, which will lead to a push notification being sent to all clients.
+
+Clients will fetch the list of latest changed collections from monitor/changes, and since they will use the bumped timestamp in the `?_expected={}` query param, they will bust the CDN cache and obtain the latest version.
+
+When iterating the list of collections and their respective data timestamp, they will compare their local data timestamp and the remote one, and will issue a synchronization if they differ.
+
+Clients will fetch the changesets using the bumped timestamp:
+
+`GET /buckets/main/collections/{cid}/changeset?_since={last-data-timestamp)&_expected={bumped-timestamp}`
+
+The response will contain the refreshed signature, and an empty list of changes give that the data itself wasn't changed since the last synchronization.
+
+**Complexity**
+
+Low. The overall architecture would not be modified.
+
+**Impact**
+
+High.
+
+Bumping the data timestamp on signature refresh will have a significant impact both on servers and on clients.
+
+We refresh the signature of all collections that haven't been updated in the last 7 days. With this option, we will force synchronization on all clients for these collections every at most 7 days.
+
+This will increase trafic on our servers, although the impact will be relatively low on the origins.
+
+> Note: traffic on origin shouldn't be too affected. And we are very likely to refresh signatures of several collections are once, therefore grouping them in one push notification.
+
+During synchronization, the clients will apply the empty list of changes (delta since last sync), and revalidate the signature of their local data.
+
+This will have an impact on clients CPUs on every signature refresh. But this will indirectly fix [bug 1640126](https://bugzilla.mozilla.org/show_bug.cgi?id=1640126), which consists in verifying local data when up-to-date.
+
+> Note: We could adjust the [behaviour of the lambda](https://github.com/mozilla-services/remote-settings-lambdas/blob/443117cb16cc9d720c59341c6c811e7bf127b289/commands/refresh_signature.py#L78-L93) to control the number of refreshed collections in one run.
```

Comment:
How much of an impact do you think this would have? Would it mitigate some of the points you made above such that it would drop the impact to a medium or a low?

Reply:
We have like 20-30 collections that weren't updated in the last 6 months. It's hard to tell how many are actually instantiated in clients, and whose signature refresh would imply initiating a synchronization. But if we keep the threshold of refreshing every 7 days at most, we could refresh 4-5 collections a day, which would definitely have a low impact on clients.